src/nft/components/profile/list/ListingButton.tsx

```diff
@@ -0,0 +1,144 @@
+import { Plural, t, Trans } from '@lingui/macro'
+import { BaseButton } from 'components/Button'
+import ms from 'ms.macro'
+import { BelowFloorWarningModal } from 'nft/components/profile/list/Modal/BelowFloorWarningModal'
+import { useIsMobile, useNFTList, useSellAsset } from 'nft/hooks'
+import { Listing, ListingStatus, WalletAsset } from 'nft/types'
+import { useEffect, useMemo, useState } from 'react'
+import styled from 'styled-components/macro'
+import { BREAKPOINTS } from 'theme'
+import shallow from 'zustand/shallow'
+
+import { getListings } from '../../bag/profile/utils'
+
+const BELOW_FLOOR_PRICE_THRESHOLD = 0.8
+
+const StyledListingButton = styled(BaseButton)<{ showResolveIssues: boolean; missingPrices: boolean }>`
+  background: ${({ showResolveIssues, theme }) => (showResolveIssues ? theme.accentFailure : theme.accentAction)};
+  color: ${({ theme }) => theme.accentTextLightPrimary};
+  font-weight: 600;
+  font-size: 20px;
+  line-height: 24px;
+  padding: 16px;
+  border-radius: 12px;
+  width: min-content;
+  border: none;
+  cursor: ${({ missingPrices }) => (missingPrices ? 'auto' : 'pointer')};
+  opacity: ${({ showResolveIssues, missingPrices }) => (showResolveIssues ? '1' : missingPrices && '0.3')};
+
+  @media screen and (max-width: ${BREAKPOINTS.sm}px) {
+    font-size: 16px;
+    line-height: 20px;
+    padding: 10px 12px;
+  }
+`
+
+export const ListingButton = ({ onClick }: { onClick: () => void }) => {
+  const { sellAssets, showResolveIssues, toggleShowResolveIssues, issues, setIssues } = useSellAsset(
+    ({ sellAssets, showResolveIssues, toggleShowResolveIssues, issues, setIssues }) => ({
+      sellAssets,
+      showResolveIssues,
+      toggleShowResolveIssues,
+      issues,
+      setIssues,
+    }),
+    shallow
+  )
+  const { setListingStatus, setListings, setCollectionsRequiringApproval } = useNFTList(
+    ({ setListingStatus, setListings, setCollectionsRequiringApproval }) => ({
+      setListingStatus,
+      setListings,
+      setCollectionsRequiringApproval,
+    }),
+    shallow
+  )
+  const [showWarning, setShowWarning] = useState(false)
+  const isMobile = useIsMobile()
+
+  // instantiate listings and collections to approve when user's modify input data
+  useEffect(() => {
+    const [newCollectionsToApprove, newListings] = getListings(sellAssets)
+    setListings(newListings)
+    setCollectionsRequiringApproval(newCollectionsToApprove)
+    setListingStatus(ListingStatus.DEFINED)
+  }, [sellAssets, setCollectionsRequiringApproval, setListingStatus, setListings])
+
+  // Find issues with item listing data
+  const [listingsMissingPrice, listingsBelowFloor] = useMemo(() => {
```

Comment:
some of these hooks might belong as their own hook -  might help readability of this file

Reply:
I'm saving hook/listing logic refactoring for a subsequent PR